ddl/rollingback.go

```diff
@@ -190,6 +190,21 @@ func rollingbackDropSchema(t *meta.Meta, job *model.Job) error {
 	return nil
 }
 
+func rollingbackTruncateTable(t *meta.Meta, job *model.Job) (ver int64, err error) {
+	tblInfo, _, err := checkTruncateTableExist(t, job)
+	if err != nil {
+		return ver, errors.Trace(err)
+	}
+
+	// Here truncate table is done in a transaction, if the job is not completed, it can be canceled.
+	if tblInfo.State == model.StatePublic {
```

Comment:
you should judge the job.SchemaState, Truncate table will never change the tblInfo.State.